### DISCUSSION TITLE
Code editor as property editor

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco-CMS#15262] [Features and ideas]
I wonder if the code editor (Ace editor) should be a property editor in core. In the new backoffice Monaco editor.

E.g. if adding a HTML block we can use a textarea to enter iframe or other third party code.
However content editors doesn't get much syntax highlighting here.

An example from Elementor:
https://elementor.com/help/html-widget/

One has to add a custom property editor or use e.g. Contentment package which includes a code editor:
https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/code-editor.md

But I think it would be useful in core since the code editor already exist and we have a Markdown editor.

**Comment by leekelleher** [comment]
+1 _(it'd be one less thing for me to support in Contentment)_.
Although I have _(provisionally)_ ported it over (in Contentment) for Bellissima: https://github.com/leekelleher/umbraco-contentment/commit/e5f63a904127e2bd0f08d89d04461f608d73d2e2

**Reply by bjarnef** [reply]
@iOvergaard @nielslyngsoe I think this would be great to consider in new backoffice too.
